util/src/main/java/org/folio/reservoir/util/readstream/XmlFixerMapper.java

```diff
@@ -0,0 +1,119 @@
+package org.folio.reservoir.util.readstream;
+
+import io.vertx.core.buffer.Buffer;
+
+public class XmlFixerMapper implements Mapper<Buffer, Buffer> {
+
+  private static final String REPLACEMENT_CHAR = "&#xFFFD;";
+
+  private boolean ended;
+
+  Buffer result;
+
+  Buffer pending = null;
+
+  private int numberOfFixes = 0;
+
+  int front;
+
+  int tail;
+
+  @Override
+  public void push(Buffer buffer) {
+    Buffer input;
+    if (result == null) {
+      result = Buffer.buffer();
+    }
+    // In most cases pending is null; especially for large buffers
+    if (pending == null) {
+      input = buffer;
+    } else {
+      pending.appendBuffer(buffer);
+      input = pending;
+    }
+    tail = 0;
+    for (front = 0; front < input.length(); front++) {
+      byte leadingByte = input.getByte(front);
+      if (leadingByte < 32 && leadingByte != 9 && leadingByte != 10 && leadingByte != 13) {
```

Comment:
We don't need to care about surrogate blocks, right?

Reply:
I dont know what you mean

Reply:
These will never be part of a multi byte sequence.  Multibyte are all >= 128.